atlas-spec/mpc-engine/src/party.rs

```diff
@@ -152,40 +183,102 @@ impl Party {
+        } = channel_msg
+        {
+            let first_msg = my_channel.recv().unwrap();
+            if let SubMessage::OTCommit(commitment) = first_msg {
+                let dst = format!("OT-{}-{}", from, to);
+                self.log(&format!("Choose left input: {choose_left}"));
+                let (receiver, resp) = crate::primitives::ot::OTReceiver::select(
+                    &mut self.entropy,
+                    dst.as_bytes(),
+                    commitment,
+                    choose_left,
+                )?;
+                their_channel.send(SubMessage::OTSelect(resp)).unwrap();
+                let second_msg = my_channel.recv().unwrap();
+                if let SubMessage::OTSend(payload) = second_msg {
+                    assert_eq!(self.id, to);
+                    let result = receiver.receive(payload)?;
+                    self.log(&format!("Got message {result:?}"));
+                    Ok(result)
+                } else {
+                    Err(Error::UnexpectedSubprotocolMessage(second_msg))
                 }
+            } else {
+                Err(Error::UnexpectedSubprotocolMessage(first_msg))
             }
-            _ => panic!("Unexpected channel message in OT"),
-        })
+        } else {
+            Err(Error::UnexpectedMessage(channel_msg))
+        }
+    }
+
+    /// Initiate an equality check with another party.
+    ///
+    /// The initiator has to provide its own input value to the check and will
+    /// learn whether that value is the same as the responders.
+    fn eq_initiate(&mut self, i: usize, my_value: &[u8]) -> Result<bool, Error> {
+        let (own_sender, own_receiver) = mpsc::channel::<SubMessage>();
+        let (their_sender, their_receiver) = mpsc::channel::<SubMessage>();
+
+        let channel_msg = Message {
+            from: self.id,
+            to: i,
+            payload: MessagePayload::SubChannel(own_sender, their_receiver),
+        };
+        self.channels.parties[i].send(channel_msg).unwrap();
+
+        let dst = format!("EQ-{}-{}", self.id, i);
+        let (commitment, opening) = Commitment::new(my_value, dst.as_bytes(), &mut self.entropy)?;
+        their_sender.send(SubMessage::EQCommit(commitment)).unwrap();
+
+        let responder_message = own_receiver.recv().unwrap();
+        if let SubMessage::EQResponse(their_value) = responder_message {
+            let res = their_value == my_value;
+            their_sender
+                .send(SubMessage::EQOpening(my_value.to_vec(), opening))
+                .unwrap();
+            Ok(res)
+        } else {
+            Err(Error::UnexpectedSubprotocolMessage(responder_message))
+        }
+    }
+
+    /// Listen for an equality check initiation.
+    ///
+    /// The responder has to provide its own input value to the check and will
+    /// learn whether that value is the same as the initators.
+    fn eq_respond(&mut self, my_value: &[u8]) -> Result<bool, Error> {
```

Comment:
```suggestion
    /// learn whether that value is the same as the initiators.
```

Reply:
I don't understand the suggestion, I'm afraid.